install/kubernetes/cilium/templates/etcd-operator-serviceaccount.yaml

```diff
@@ -1,11 +0,0 @@
-{{- if .Values.etcd.managed }}
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: cilium-etcd-sa
```

Comment:
This looks unused, but it's actually not. There are references to it here (different repository, but deployed as part of the `cilium-etcd-operator`):
https://github.com/cilium/cilium-etcd-operator/blob/fdf1993dded89db77742916a9e55ef898a4e5ad0/etcd-operator/descriptors.go#L62

Cilium also has a a well-known identity for this SA as well:
https://github.com/cilium/cilium/blob/4f27865e975cc095891b68df5f4a7bc0e26ecca5/pkg/identity/numericidentity.go#L171-L176

I think we want to remove managed-etcd in Cilium 1.11 (see https://github.com/cilium/cilium/pull/15464/), as it's already deprecated. So this file here will have to go, but there is more that than just deleting this file. We also need to
- Remove the `cilium-etcd-operator` deployment from the helm chart
- Remove all references to `.Values.etcd.managed`
- Mention the removal in the upgrade guide
- Ensure there are no backwards-compatibility issues

Might be worth doing the `etcd.managed` removal in a separate PR, so we have separation of concerns and better code reviews.

Reply:
Thank you for clarify. I changed it back and let `etcd.managed` removal in another PR :100: